nni/retiarii/nn/pytorch/mutator.py

```diff
@@ -116,12 +120,94 @@ def process_inline_mutation(model: Model) -> Optional[List[Mutator]]:
+        else:
+            return node.operation.parameters['candidates']
+
+    @staticmethod
+    def number_of_chosen(node):
+        if 'n_chosen' in node.operation.parameters:
+            return node.operation.parameters['n_chosen']
+        return 1
+
+    def mutate(self, model: Model):
+        # this mutate does not have any effect, but it is recorded in the mutation history
+        for node in model.get_nodes_by_label(self.label):
+            for _ in range(self.number_of_chosen(node)):
+                self.choice(self.candidates(node))
+            break
+
+
+def extract_mutation_from_pt_module(pytorch_model: nn.Module) -> Tuple[Model, Optional[List[Mutator]]]:
+    model = Model(_internal=True)
+    graph = Graph(model, uid(), '_model', _internal=True)._register()
+    model.python_class = pytorch_model.__class__
+    if len(inspect.signature(model.python_class.__init__).parameters) > 1:
+        if not hasattr(pytorch_model, '_init_parameters'):
+            raise ValueError('Please annotate the model with @serialize decorator in python execution mode '
+                             'if your model has init parameters.')
+        model.python_init_params = pytorch_model._init_parameters
+    else:
+        model.python_init_params = {}
+
+    for name, module in pytorch_model.named_modules():
+        # tricky case: value choice that serves as parameters are stored in _init_parameters
+        if hasattr(module, '_init_parameters'):
+            for key, value in module._init_parameters.items():
+                if isinstance(value, ValueChoice):
+                    node = model.root_graph.add_node(name + '.init.' + key, 'ValueChoice', {'candidates': value.candidates})
+                    node.label = value.label
+
+        if isinstance(module, (LayerChoice, InputChoice, ValueChoice)):
+            # TODO: check the label of module and warn if it's auto-generated
+            pass
+        if isinstance(module, LayerChoice):
+            node = model.root_graph.add_node(name, 'LayerChoice', {'candidates': module.names})
+            node.label = module.label
+        if isinstance(module, InputChoice):
+            node = model.root_graph.add_node(name, 'InputChoice',
+                                             {'n_candidates': module.n_candidates, 'n_chosen': module.n_chosen})
+            node.label = module.label
+        if isinstance(module, ValueChoice):
+            node = model.root_graph.add_node(name, 'ValueChoice',
+                                             {'candidates': module.candidates})
+            node.label = module.label
+        if isinstance(module, Placeholder):
+            raise NotImplementedError('Placeholder is not supported in python execution mode.')
+
+    model.status = ModelStatus.Frozen
+    if not graph.hidden_nodes:
+        return model, None
+
+    mutators = []
+    for nodes in _group_by_label_and_type(graph.hidden_nodes):
```

Comment:
it is a little strange that both root_graph and graph are used, they are the same graph?

Reply:
I think so.